lib/node_modules/@stdlib/ndarray/any-by/lib/main.js

```diff
@@ -180,7 +180,7 @@ function anyBy( x, options, predicate, thisArg ) {
 
 	// Check whether we need to reinsert singleton dimensions which can be useful for broadcasting the returned output array to the shape of the original input array...
 	if ( opts.keepdims ) {
-		y = spreadDimensions( N, y, idx );
+		y = spreadDimensions( N, y, idx, false );
```

Comment:
```suggestion
		y = spreadDimensions( N, y, idx, true );
```

I just realized that we should actually be returning `writable` instances here, as that was the behavior previously. Previously, only if an array was "read-only" did we return a read-only array. Here, `y` was allocated as a writable array. Hence, we should be returning a writable array, not a read-only array.